### PR TITLE
Feature/elprep

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/CenterForMedicalGeneticsGhent/nf-cmgg-preprocessing/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the CenterForMedicalGeneticsGhent/nf-cmgg-preprocessing _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by matthdsm** [comment]
This one should work but has a filename collision in `ELPREP`, we should fix `$prefix` for the aligner process

File: subworkflows/local/fastq_to_cram/main.nf [hunk 1]
```diff
@@ -119,13 +125,13 @@ def gather_split_files_per_sample(ch_files) {
         meta, files ->
         return [
             groupKey(
-                meta - meta.subMap('id', 'readgroup', 'count') + [id: meta.samplename],
-                meta.count.toInteger()
+                meta - meta.subMap('id', 'readgroup', 'chunks', 'lane') + [id: meta.samplename],
+                meta.count ?: 1 * meta.chunks?: 1
```
Comment (nvnieuwk) [inline subworkflows/local/fastq_to_cram/main.nf:129]:
```suggestion
                meta.count ?: 1 * meta.chunks ?: 1
```
Weird this doesn't give an error?

File: subworkflows/local/fastq_to_cram/main.nf [hunk 2]
```diff
@@ -119,13 +125,13 @@ def gather_split_files_per_sample(ch_files) {
...
             ),
             files
         ]
     }
-    .groupTuple( by: [0])
+    .groupTuple(by:[0])
```
Comment (nvnieuwk) [inline subworkflows/local/fastq_to_cram/main.nf:134]: No group size here?

**Comment by nvnieuwk** [comment]
Also adding a test for elprep and bamsormadup would be good I think?

File: subworkflows/local/fastq_to_cram/main.nf (hunk 2 above)
Comment (matthdsm) [reply subworkflows/local/fastq_to_cram/main.nf:134]: dammit! missed that one

Comment (matthdsm) [reply subworkflows/local/fastq_to_cram/main.nf:134]: size is in the groupKey!

Comment (nvnieuwk) [reply subworkflows/local/fastq_to_cram/main.nf:134]: Yep I'm blind!

**Review by nvnieuwk — APPROVED** [review]
LGTM then!

**Pull request merged**